exporter/signalfxexporter/exporter.go

```diff
@@ -56,11 +56,18 @@ func (sme *signalfMetadataExporter) ConsumeMetadata(metadata []*metadata.Metadat
 	return sme.pushMetadata(metadata)
 }
 
-type signalfxExporter struct {
+type metricExporter struct {
 	pushMetricsData    func(ctx context.Context, md pdata.Metrics) (droppedTimeSeries int, err error)
 	pushMetadata       func(metadata []*metadata.MetadataUpdate) error
-	pushLogsData       func(ctx context.Context, ld pdata.Logs) (droppedLogRecords int, err error)
 	hostMetadataSyncer *hostmetadata.Syncer
+	config             *Config
+	logger             *zap.Logger
+}
+
+type logExporter struct {
```

Comment:
why are this division and associated updates necessary?

Reply:
The end goal here is to use [HTTPClientSettings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L36) to make 'httpClient' instead of creating it in the same code. In order to do that, we need a host of type component.Host in [httpClient.ToClient()](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L99
) method which can only be available in a [Start](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L90
) method AFAIK. So, The approach we decided to use was to create a Start method and call newSignalFxExporter or newEventExporter in the [Start](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L90
) method based on the context of the [Start](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L90
) method (from [createMetricsExporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/factory.go#L99) or [createLogsExporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/factory.go#L179
)).
If we use the same structure for Logs as well as Metrics, we have two potential solutions:
1. Call both method ([newSignalFxExporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L95) or [newEventExporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L184
)) in the [Start](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L90
) regardless of the context(createMetricsExporter or createLogsExporter).
2. Identify the context of the [Start](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L90
) method and call the respective function from the Start method.

In the first solution, it does not seem optimal as we would be calling an unnecessary function (newSignalFxExporter or newEventExporter) from the [Start](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L90
) method. For ex., If a user configures only one pipeline (metrics) still newEventExporter will be called which is unnecessary.

In the second solution, we need to identify the context of the [Start](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L90
) method. Using the existing signalfxExporter structure, it seems unlikely to identify the context of the [Start](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b96dfffb063e33f771425a54f65afbf9741dd482/exporter/signalfxexporter/exporter.go#L90
) method (createMetricsExporter or createLogsExporter).

Hence, we used different structures for the Logs and Metrics pipeline. So that at runtime, we will have different Start methods for different pipelines and we can get context.

Reply:
I don't believe this is the correct approach, and the ask is to embed the `confighttp.HTTPClientSettings` into the exporter config for usage in client instantiation and not create a new instance directly. Briefly walking through this none of the changes you're making appear necessary.  We aren't using custom authenticators so passing in an empty extension map to `ToClient(map[config.ComponentID]component.Extension{}`)  should be ok afaict.

For example:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/zipkinexporter/config.go#L31
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/jaegerthrifthttpexporter/config.go#L25

It's important that these settings are backward compatible for field renames so legacy fields should be used to set the client setting field values if applicable.

Reply:
Added `confighttp.HTTPClientSettings` into exporter config and changed `MaxConnections` to pointer for backward compatibility.